snapcraft/storeapi/__init__.py

```diff
@@ -292,7 +292,11 @@ def _download_snap(self, name, channel, arch, download_path,
         not_downloaded = True
         retry_count = 5
         while not_downloaded and retry_count:
-            request = self.cpi.get(download_url, stream=True)
+            headers = {}
+            if os.path.exists(download_path):
+                headers['Range'] = 'StartPos: %d'.format(
+                    os.path.getsize(download_path))
```

Comment:
Internap has `Accept-Ranges: bytes`, so we should be using `Range: bytes=<size>-'`.

See the full workflow in action with curl in https://pastebin.canonical.com/189327/